src/test/scala/cognite/spark/v1/FileContentRelationTest.scala

```diff
@@ -0,0 +1,248 @@
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("externalId", fileExternalId)
+      .option("inferSchema", true)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    val result = spark.sqlContext.sql(s"select * from filecontent").collect()
+    result.map(_.toSeq.toList) should contain theSameElementsAs
+      Array(
+        List(30, "Alice", null),
+        List(25, "Bob", null),
+        List(35, "Charlie", null),
+        List(35, "Charlie2", "test")
+      )
+  }
+
+  it should "fail with a sensible error if the mimetype is wrong" in {
+    val exception = sparkIntercept {
+      val sourceDf: DataFrame = dataFrameReaderUsingOidc
+        .useOIDCWrite
+        .option("type", "filecontent")
+        .option("externalId", fileWithWrongMimeTypeExternalId)
+        .load()
+      sourceDf.createOrReplaceTempView("fileContent")
+      spark.sqlContext.sql(s"select * from filecontent").collect()
+    }
+    assert(exception.getMessage.contains("Wrong mimetype. Expects application/jsonlines"))
+  }
+
+  it should "infer the schema" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalId)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+      StructField("name", StringType, nullable = true),
+      StructField("test", StringType, nullable = true),
+      StructField("age", LongType, nullable = true)
+    )
+  }
+
+  it should "infer the schema correctly with nested json" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalIdWithNestedJson)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+      StructField("name", StringType, nullable = true),
+      StructField("test",
+        ArrayType(
+          StructType(
+            Seq(
+              StructField("key",StringType, nullable = true),
+              StructField("key2",StringType, nullable = true)
+            )
+          ),
+          containsNull = true
+        ),
+        nullable = true
+      ),
+      StructField("age", LongType, nullable = true)
+    )
+  }
+
+  it should "infer the schema correctly with conflicting types" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalIdWithConflicts)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
```

Comment:
what do conflicts looks like and what happens to them, could there be an assert on that? or at least extra comment

Reply:
what do you mean? There is a test with conflicting schema, and string takes over.

Reply:
"infer the schema correctly with conflicting types" is the test

Reply:
ouch sorry, I was sure when I looked at it that it was checking rows contents :facepalm: probably lines from other tests somehow were in view